### PR TITLE
Cleaned up promises to use established patterns

### LINKED ISSUE #144: Clean up Promise logic to use established patterns
We use a lot of `new Promise<>((resolve, reject) => {})` when we can get rid of some of those in favor of `Promise.resolve(...)` and `Promise.reject(...)`.

### PR DESCRIPTION
**Pull request by mttwc**
Fixes #144 

@VishaalK I'll need your expert eyes on this one :P

### CODE REVIEW

File: src/scripts/clipperUI/components/sectionPicker.tsx [hunk 1]
```diff
@@ -210,16 +210,12 @@ export class SectionPickerClass extends ComponentBase<SectionPickerState, Sectio
...
-			SectionPickerClass.dataSource.getNotebooks(headers).then((responsePackage: OneNoteApi.ResponsePackage<OneNoteApi.Notebook[]>) => {
-				resolve(responsePackage);
-			}, (failure: OneNoteApi.RequestError) => {
-				reject(failure);
-			});
+		let headers: { [key: string]: string } = {};
+		headers[Constants.HeaderValues.appIdKey] = Settings.getSetting("App_Id");
+		headers[Constants.HeaderValues.userSessionIdKey] = sessionId;
+
+		return SectionPickerClass.dataSource.getNotebooks(headers).catch((failure: OneNoteApi.RequestError) => {
```
Comment (VishaalK) [inline src/scripts/clipperUI/components/sectionPicker.tsx]: The `catch` is unnecessary here. Returning the promise will propagate the error already.

Comment (VishaalK) [reply src/scripts/clipperUI/components/sectionPicker.tsx]: `sectionPicker.tsx` has the error handler its in `.then()` to use the error that is returned

File: src/scripts/saveToOneNote/oneNoteSaveablePdf.ts [hunk 2]
```diff
@@ -22,10 +22,8 @@ export class OneNoteSaveablePdf implements OneNoteSaveable {
 	}
 
 	public getPatch(index: number): Promise<OneNoteApi.Revision[]> {
-		return new Promise<OneNoteApi.Revision[]>((resolve) => {
-			this.pdf.getPageListAsDataUrls(this.buckets[index]).then((dataUrls) => {
-				resolve(this.createPatchRequestBody(dataUrls));
-			});
+		return this.pdf.getPageListAsDataUrls(this.buckets[index]).then((dataUrls) => {
```
Comment (VishaalK) [inline src/scripts/saveToOneNote/oneNoteSaveablePdf.ts]: I think you can do this here:

`return this.pdf.getPageListAsDataUrls(this.buckets[index]).then(this.createPatchRequestBody);`

It's debatable whether or not this is more readable though.

File: src/scripts/localization/localizationHelper.ts [hunk 3]
```diff
@@ -10,15 +10,13 @@ export class LocalizationHelper {
...
-					parsedResponse: request.responseText
-				});
-			}, (error) => {
-				reject(error);
+		return HttpWithRetries.get(url).then((request) => {
+			return Promise.resolve({
+				request: request,
+				parsedResponse: request.responseText
 			});
+		}).catch((error) => {
```
Comment (VishaalK) [inline src/scripts/localization/localizationHelper.ts]: I think this `catch` is also unnecessary

File: src/scripts/http/cachedHttp.ts [hunk 4]
```diff
@@ -68,19 +63,17 @@ export class CachedHttp {
...
-			getRemoteValue().then((responsePackage) => {
-				let setValue: TimeStampedData = this.setTimeStampedValue(key, responsePackage.parsedResponse);
-				if (!setValue) {
-					// Fresh data from the remote was unavailable
-					this.cache.removeKey(key);
-				}
-				resolve(setValue);
-			}, (error: OneNoteApi.RequestError) => {
-				// TODO: Don't use OneNoteApi.RequestError
+		return getRemoteValue().then((responsePackage) => {
```
Comment (VishaalK) [inline src/scripts/http/cachedHttp.ts:64]: As a side note, why does `getRemoteValue()` not take any arguments?

File: src/scripts/contentCapture/augmentationHelper.ts [hunk 5]
```diff
@@ -103,33 +102,31 @@ export class AugmentationHelper {
...
+					return Promise.reject(OneNoteApi.ErrorUtils.createRequestErrorObject(request, OneNoteApi.RequestErrorType.UNABLE_TO_PARSE_RESPONSE));
+				}
+
+				let responsePackage = {
+					parsedResponse: parsedResponse,
+					request: request
+				};
+				return Promise.resolve(responsePackage);
+			}).catch((error: OneNoteApi.RequestError) => {
+				return Promise.reject(error);
```
Comment (VishaalK) [inline src/scripts/contentCapture/augmentationHelper.ts]: This `catch` is unnecessary

File: src/scripts/contentCapture/bookmarkHelper.ts [hunk 6]
```diff
@@ -65,99 +65,97 @@ export class BookmarkHelper {
...
-				resolve(result);
-			}
-		});
+				return Promise.resolve(result);
+			});
+		} else {
+			bookmarkPageEvent.setCustomProperty(Log.PropertyName.Custom.BookmarkInfo, JSON.stringify(bookmarkLoggingInfo));
+
+			Clipper.logger.logEvent(bookmarkPageEvent);
+			return Promise.resolve(result);
```
Comment (VishaalK) [inline src/scripts/contentCapture/bookmarkHelper.ts:157]: Is this all one method? The promise stuff looks fine though

File: src/scripts/extensions/extensionBase.ts [hunk 7]
```diff
@@ -188,15 +188,13 @@ export abstract class ExtensionBase<TWorker extends ExtensionWorkerBase<TTab, TT
...
-			}, (error: OneNoteApi.GenericError) => {
-				reject(error);
-			});
+		return this.clipperData.getFreshValue(ClipperStorageKeys.flightingInfo, fetchNonLocalData, Experiments.updateIntervalForFlights).then((successfulResponse) => {
+			// The response comes as a string array in the form [flight1, flight2, flight3],
+			// needs to be in CSV format for ODIN cooker, so we rejoin w/o spaces
+			let parsedResponse: string[] = successfulResponse.data.Features ? successfulResponse.data.Features : [];
+			return Promise.resolve(parsedResponse);
+		}).catch((error: OneNoteApi.GenericError) => {
+			return Promise.reject(error);
```
Comment (VishaalK) [inline src/scripts/extensions/extensionBase.ts]: Unnecessary `catch`.

**Review by VishaalK — CHANGES_REQUESTED** [review]
Had some minor comments. Looks dope!

File: src/scripts/saveToOneNote/oneNoteSaveablePdf.ts (hunk 2 above)
Comment (mttwc) [reply src/scripts/saveToOneNote/oneNoteSaveablePdf.ts]: Interesting! I kinda like this syntax. It's succinct.

File: src/scripts/http/cachedHttp.ts (hunk 4 above)
Comment (mttwc) [reply src/scripts/http/cachedHttp.ts:64]: getRemoteValue is passed in as a parameter and defines how the remove value should be gotten.

File: src/scripts/contentCapture/bookmarkHelper.ts (hunk 6 above)
Comment (mttwc) [reply src/scripts/contentCapture/bookmarkHelper.ts:157]: Yep it is. It's pretty big so I'm sure we can refactor.

File: src/scripts/clipperUI/components/sectionPicker.tsx (hunk 1 above)
Comment (mttwc) [reply src/scripts/clipperUI/components/sectionPicker.tsx]: Resolved

File: src/scripts/localization/localizationHelper.ts (hunk 3 above)
Comment (mttwc) [reply src/scripts/localization/localizationHelper.ts]: Resolved

File: src/scripts/contentCapture/augmentationHelper.ts (hunk 5 above)
Comment (mttwc) [reply src/scripts/contentCapture/augmentationHelper.ts]: Resolved

File: src/scripts/extensions/extensionBase.ts (hunk 7 above)
Comment (mttwc) [reply src/scripts/extensions/extensionBase.ts]: Resolved

**Comment by mttwc** [comment]
@VishaalK I just pushed up another commit if you want to review that.

**Review by VishaalK — APPROVED** [review]

**Pull request merged**